### PR TITLE
DO NOT MERGE: Remove all only for review

### PR DESCRIPTION
**Pull request by krishnangovindraj**
PR for review

### CODE REVIEW

File: json-test.java [hunk 1]
```diff
@@ -1,246 +0,0 @@
...
-import com.vaticle.typedb.client.api.TypeDBSession;
-import com.vaticle.typedb.client.api.TypeDBTransaction;
-import com.vaticle.typedb.client.TypeDB;
-import com.vaticle.typeql.lang.TypeQL;
-
-import static com.vaticle.typeql.lang.TypeQL.*;
-
-import com.vaticle.typeql.lang.query.TypeQLInsert;
-
-class json_test
```
Comment (krishnangovindraj) [inline json-test.java]: Rename this class and file to something more descriptive, such as `DataLoader`
The convention of Java for class names is to capitalise the first letter of each word.

(Use the IDE refactor->rename feature that you get when you right click it)

File: json-test.java [hunk 2]
```diff
@@ -1,246 +0,0 @@
...
-{
-    public static int get_id(Json file, int idx)
```
Comment (krishnangovindraj) [inline json-test.java]: Everything apart from classes (functions, variables, etc) are typically with camelCase, where you capitalise the first letter of each word **except the very first letter** - here `getId`
(Like in python, which I guess is what you've followed so far)

File: json-test.java [hunk 3]
```diff
@@ -1,246 +0,0 @@
...
-    {
-        return file.at("nodes").at(idx).at("id").asInteger();
-    }
-    public static String get_name(Json file, int idx)
-    {
-        return file.at("nodes").at(idx).at("name").asString();
-    }
-    public static String get_school(Json file, int idx)
```
Comment (krishnangovindraj) [inline json-test.java]: Have one line between functions.

File: json-test.java [hunk 4]
```diff
@@ -1,246 +0,0 @@
...
-    {
-        return  file.at("nodes").at(idx).at("school").asString();
-    }
-    public static int get_year(Json file, int idx)
-    {
-        if (!file.at("nodes").at(idx).at("year").isNull())
-        return file.at("nodes").at(idx).at("year").asInteger();
```
Comment (krishnangovindraj) [inline json-test.java]: Try to indent the `return` statement (or use braces) if it's the true-case of the `if`.

File: json-test.java [hunk 5]
```diff
@@ -1,246 +0,0 @@
...
-                //first purge existing data
-                try (TypeDBTransaction writeTransaction = session.transaction(TypeDBTransaction.Type.WRITE))
-                {
-                    TypeQLDelete query = TypeQL.match(
-                            var("p").isa("thing")
-                    ).delete(var("p").isa("thing"));
-                    writeTransaction.query().delete(query);
-                    // to persist changes, a write-transaction must always be committed (closed)
-                    writeTransaction.commit();
-                }
```
Comment (krishnangovindraj) [inline json-test.java]: Just use the database delete & create functions: https://docs.vaticle.com/docs/client-api/java#check-if-a-database-exists

File: json-test.java [hunk 6]
```diff
@@ -1,246 +0,0 @@
...
-                        TypeQLInsert insertQuery = TypeQL.insert
-                                (var("x").isa("mathematician")
-                                        .has("id", get_id(file, z))
-                                        .has("name", get_name(file, z))
-                                        .has("year_of_degree", get_year(file, z))
-                                );
-                        writeTransaction.query().insert(insertQuery);
-                        // to persist changes, a write-transaction must always be committed (closed)
-                        writeTransaction.commit();
-                        System.out.println("mathematician instantiated");
```
Comment (krishnangovindraj) [inline json-test.java]: Will be very noisy if you print for each one. Can you keep a count and print at the end or every 100

File: json-test.java [hunk 7]
```diff
@@ -1,246 +0,0 @@
...
-                        }
-                    }
-                }
-            }
-        }
-        catch(IOException e)
-        {
-            System.out.println("IO Exception in file: "+e.getMessage());
-            e.printStackTrace();
-        }
```
Comment (krishnangovindraj) [inline json-test.java]: Can the IOException be thrown from anywhere in the code? I expect it to be only the line which reads the file.
Try to keep the try-catch block as small as possible. In this case, I'd just return/exit if the file-loading failed.

File: json-test.java [hunk 8]
```diff
@@ -1,246 +0,0 @@
...
-                {
-                    TypeQLDelete query = TypeQL.match(
-                            var("p").isa("thing")
-                    ).delete(var("p").isa("thing"));
-                    writeTransaction.query().delete(query);
-                    // to persist changes, a write-transaction must always be committed (closed)
-                    writeTransaction.commit();
-                }
-
-                //for instantiating mathematicians
```
Comment (krishnangovindraj) [inline json-test.java]: Can we move each of these sections to a function so the main isn't this long to navigate?
You should be able to do it and just pass the session and file to each function.

e.g. `instantiateMathematicians(...)`

File: DataLoaderTest.java [hunk 9]
```diff
@@ -1,123 +0,0 @@
...
-                TypeQLMatch.Unfiltered.Aggregate query=TypeQL.match(var("x").isa("subject")).count();
-                QueryFuture<Numeric> ans=transaction.query().match(query);
-                assertEquals(ans.get().asLong(), 11);
-            }
-            //2 co-teach
-            {
-                TypeQLMatch.Unfiltered.Aggregate query=TypeQL.match(var("x").isa("co-teach")).count();
-                QueryFuture<Numeric> ans=transaction.query().match(query);
-                assertEquals(ans.get().asLong(), 2);
-            }
```
Comment (krishnangovindraj) [inline DataLoaderTest.java]: This should work ok, but I've just realised we don't test edges.
You could figure out a query to make sure the role-player edges and has edges are properly inserted.

Comment (krishnangovindraj) [inline DataLoaderTest.java]: I would personally run a single query which ` match $x isa! $type; group $type; count;` and assert results. It would be a lot more concise.

File: DataLoaderTest.java [hunk 10]
```diff
@@ -1,123 +0,0 @@
...
-        }
-    }
-
-    @After
-    public void disconnect()
-    {
-        session.close();
-        client.close();
-    }
-}
```
Comment (krishnangovindraj) [inline DataLoaderTest.java]: This little no-entry sign indicates that you're missing a new line at the end of the file. It's convention to have one.

File: BUILD [hunk 11]
```diff
@@ -1,23 +0,0 @@
...
-        ]
-)
-java_test(
-    name="DataLoaderTest",
-    test_class="com.vaticle.typedb.example.genealogy.math_genealogy.DataLoaderTest",
-    srcs=["DataLoaderTest.java"],
-    deps = ["@maven//:org_sharegov_mjson","@vaticle_typedb_client_java//:client-java",
-                                                                           "@vaticle_typedb_client_java//api",
-                                                                         "@vaticle_typeql//java:typeql-lang",
-                                                                         "@vaticle_typeql//java/query", ":json_test"]
```
Comment (krishnangovindraj) [inline BUILD]: Could you fix the indenting on this?

**Review by krishnangovindraj — COMMENTED** [review]
First round - reviewed the code.

File: BUILD [hunk 12]
```diff
@@ -1,23 +0,0 @@
-java_binary(
-    name = "json_test",
```
Comment (krishnangovindraj) [inline BUILD]: By convention, we use `all-small-and-hyphens-between-words` for target names

File: BUILD [hunk 13]
```diff
@@ -1,23 +0,0 @@
...
-                                                              "@vaticle_typeql//java:typeql-lang",
-                                                              "@vaticle_typeql//java/query"],
-    data =
-        [
-            ":data.json",
-            ":datalite.json"
-        ]
-)
-java_test(
-    name="DataLoaderTest",
```
Comment (krishnangovindraj) [inline BUILD]: `data-loader-test`

File: dependencies/maven/artifacts.bzl [hunk 14]
```diff
@@ -1,45 +0,0 @@
...
-    "org.jetbrains.compose.ui:ui-desktop",
-    "org.jetbrains.compose.ui:ui-graphics-desktop",
-    "org.jetbrains.compose.ui:ui-text-desktop",
-    "org.jetbrains.compose.ui:ui-unit-desktop",
-    "org.jetbrains.kotlinx:kotlinx-coroutines-core-jvm",
-    "org.jetbrains.skiko:skiko-awt-runtime-linux-x64",
-    "org.jetbrains.skiko:skiko-awt-runtime-macos-x64",
-    "org.jetbrains.skiko:skiko-awt-runtime-windows-x64",
-    "org.kohsuke:github-api",
-    "org.sharegov:mjson",
```
Comment (krishnangovindraj) [inline dependencies/maven/artifacts.bzl]: Let's do a clean up here. You only use the json one so that should be all we need here.

File: dependencies/vaticle/artifacts.bzl [hunk 15]
```diff
@@ -1,33 +0,0 @@
...
-
-def vaticle_typedb_artifact():
-    native_artifact_files(
-        name = "vaticle_typedb_artifact",
-        group_name = "vaticle_typedb",
-        artifact_name = "typedb-server-{platform}-{version}.{ext}",
-        tag_source = deployment["artifact.release"],
-        commit_source = deployment["artifact.snapshot"],
-        tag = "2.11.1"
-    )
```
Comment (krishnangovindraj) [inline dependencies/vaticle/artifacts.bzl:33]: This reminds me, we should make the factory automation .yml
Aaaand we'll have to update this tag based on the client version you're using.

File: dependencies/vaticle/repositories.bzl [hunk 16]
```diff
@@ -1,50 +0,0 @@
...
-        name = "vaticle_typeql",
-        remote = "https://github.com/vaticle/typeql",
-        commit = "e44b3000ee297fc453d1f3a835253be3181f8ed1", # sync-marker: do not remove this comment, this is used for sync-dependencies by @vaticle_typeql_lang_java
-    )
-
-def vaticle_typedb_client_java():
-    git_repository(
-        name = "vaticle_typedb_client_java",
-        remote = "https://github.com/vaticle/client-java",
-        commit = "8c5b0ffe7c1a63613da4aaec4137de1058434fe0",  # sync-marker: do not remove this comment, this is used for sync-dependencies by @vaticle_typedb_client_java
```
Comment (krishnangovindraj) [inline dependencies/vaticle/repositories.bzl:49]: Let's see if we can use version `2.17.1` for client-java and typeql + Use their corresponding common & dependencies commits.

**Review by krishnangovindraj — COMMENTED** [review]
Some build/workspace reviews

File: json-test.java (hunk 5 above)
Comment (shiladitya-mukherjee) [reply json-test.java]: but then we also need to define the schema and the rules for the database.

Comment (krishnangovindraj) [reply json-test.java]: That's true. Just read it from a file and do a define query.

File: json-test.java (hunk 8 above)
Comment (shiladitya-mukherjee) [reply json-test.java]: done

File: DataLoaderTest.java (hunk 9 above)
Comment (shiladitya-mukherjee) [reply DataLoaderTest.java]: sorry, probably you made some typos in this comment. I cannot fully understand. Can you rephrase it?

Comment (krishnangovindraj) [reply DataLoaderTest.java]: `isa` will test the existence of entity, relation and attribute vertices in the graph.
We don't have a way of testing whether the edges between them have been inserted. 

If we have time at the end of the review, we could figure out a way to test this.

Comment (shiladitya-mukherjee) [reply DataLoaderTest.java]: on hold

File: dependencies/vaticle/artifacts.bzl (hunk 15 above)
Comment (shiladitya-mukherjee) [reply dependencies/vaticle/artifacts.bzl:33]: oops, totally forgot about the YAML file

File: BUILD (hunk 12 above)
Comment (shiladitya-mukherjee) [reply BUILD]: But Java does not allow hyphens in variable names, so if we want to call this class from other files, we wont be able to?

File: BUILD (hunk 13 above)
Comment (shiladitya-mukherjee) [reply BUILD]: But Java does not allow hyphens in variable names, so if we want to call this class from other files, we wont be able to?

File: BUILD (hunk 12 above)
Comment (krishnangovindraj) [reply BUILD]: It's the target which will be named with hyphens. It's independent of the class name

File: README.md [hunk 17]
```diff
@@ -1,218 +0,0 @@
...
-"74—Mechanics of deformable solids" isa name => 2
-"05—Combinatorics" isa name => 1
-"97—Mathematics education" isa name => 1
-"57—Manifolds and cell complexes" isa name => 3
-"13—Commutative rings and algebras" isa name => 1
-"20—Group theory and generalizations" isa name => 1
-```
-
-## Appendix
-### Schema
```
Comment (krishnangovindraj) [inline README.md]: Move this into a `schema.tql` file instead and load that in your DataMigrator.

File: json-test.java (hunk 5 above)
Comment (shiladitya-mukherjee) [reply json-test.java]: done

File: DataLoaderTest.java (hunk 9 above)
Comment (shiladitya-mukherjee) [reply DataLoaderTest.java]: done

File: README.md (hunk 17 above)
Comment (shiladitya-mukherjee) [reply README.md]: done
